scripts/create_service.sh

```diff
@@ -14,7 +14,8 @@ port="$3"
 hostname_suffix="$4"
 shift 4 # Remaining arguments ($@) are passed to the server binary.
 
-repo_version=${repo_version-"master"}
+# EXAMPLES_VERSION may be a branch name or a tag in the git repo.
+EXAMPLES_VERSION=${EXAMPLES_VERSION-"master"}
```

Comment:
If `EXAMPLES_VERSION=""` (empty string, but not unset), this ends up producing `git clone -b --single-branch --depth=1  ...` in the startup-script. Not sure if this qualifies as needing to be fixed or not, but I was debugging one deployment where the startup-script had an empty branch name and this was the only cause I could think of with that outcome. Should this check if `EXAMPLES_VERSION` is set and non-empty, or would that just lead to some other suprise/problem?

Reply:
I think the only reasonable "fix" here would be to error in this case.  It leads to a bad command-line, but I don't want to assume the user wants "master" if they've somehow set EXAMPLES_VERSION explicitly to the empty string.

Are you sure this was the root cause of a problem you saw, or was it the other thing you noticed instead (the missing `go build .` command)?